apps/web/src/libs/wallet-connect-adapter.ts

```diff
@@ -14,13 +14,19 @@ export class WalletConnectAdapter extends EventTarget implements IConnector {
     super()
   }
 
+  private async setupPeraWallet() {
+    if (!this._peraWallet) {
+      const { PeraWalletConnect } = await import('@perawallet/connect')
+      this._peraWallet = new PeraWalletConnect({
+        bridge: BRIDGE_URL,
+      })
+    }
+  }
+
   private async initialize() {
-    const { PeraWalletConnect } = await import('@perawallet/connect')
+    await this.setupPeraWallet()
+    await this._peraWallet.connect()
 
-    this._peraWallet = new PeraWalletConnect({
-      bridge: BRIDGE_URL,
-    })
-    await this._peraWallet.reconnectSession()
```

Comment:
calling seems`reconnectSession` throws an error if it is your first time connecting your wallet